src/mon/OSDMonitor.cc

```diff
@@ -4677,6 +4678,28 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
 	}
       }
       rdata.append(ds);
+    } else if (prefix == "osd info") {
+      int64_t osd_id;
+      bool do_single_osd = true;
+      if (!cmd_getval(cct, cmdmap, "id", osd_id)) {
+	do_single_osd = false;
+      }
+
+      if (f) {
+	if (do_single_osd) {
+	  osdmap.dump_osd(osd_id, f.get());
```

Comment:
i think dump_osd() will crash if you give it an osd that !exists()

Reply:
addressed, and also fixed an additional bug that came up while testing it.

added an additional guard to the osdmap functions too, in case someone calls them in the future without checking whether the osd exists.